tasks/push-to-cdn/push-to-cdn.yaml

```diff
@@ -0,0 +1,140 @@
+          EXODUS_GW_KEY="exodus-stage-privatekey.pem"
+          EXODUS_GW_URL="https://exodus-gw.corp.stage.redhat.com"
+        fi
+
+        # Check if EXODUS_ENV contains "cdn-stage"
+        if [[ "$EXODUS_ENV" == *"cdn-stage"* ]]; then
+          EXODUS_GW_ENV="pre"
+        fi
+
+        cat << EOF > $exodus_rsync_config_path
+        gwcert: $EXODUS_GW_CERT
+        gwkey: $EXODUS_GW_KEY
+        gwurl: $EXODUS_GW_URL
+        gwenv: $EXODUS_GW_ENV
+
+        logger: file:/proc/1/fd/1
+        loglevel: INFO
+
+        environments:
+        - prefix: exodus
+        EOF
+        echo "exodus-rsync.conf created here: $exodus_rsync_config_path"
+      # How to mount secrets that are in the vault here
+      # volumeMounts:
+      #   - name: secrets
+      #     mountPath: "/secrets"
+    - name: create-source-destination-for-files
+      image: quay.io/distribution_relengs/publish-to-dev-portal-test-image
+      script: |
+        #!/bin/bash
+        set -e  
+        BINARIES_PATH="$(workspaces.data.path)/$(params.binaries_dir)"
+
+        # Check if the directory exists
+        if [ ! -d "$BINARIES_PATH" ]; then
+            echo "Directory $BINARIES_PATH does not exist."
+            exit 1
+        fi
+
+        json_output="{}"
+        # Loop through each file in the directory
+        for BINARY_PATH in "$BINARIES_PATH"/*; do
+            if [ -f "$BINARY_PATH" ]; then
+                echo "Processing binary: $(basename "$BINARY_PATH")"
+                checksum=$(sha256sum "$BINARY_PATH" | awk '{print $1}')
+                first_two_chars=${checksum:0:2}
+                destination_path="exodus:/content/origin/files/sha256/$first_two_chars/$checksum/$(basename "$BINARY_PATH")"
+                echo "Destination dir: $destination_path"
+                echo ""
+                echo -n "$destination_path" >> "$(results.srcDestPath.path)"
+                # Add key:value (src:dest) pair to the json_output
+                json_output=$(echo "$json_output" | jq --arg key "$BINARY_PATH" --arg value "$destination_path" '.[$key]=$value')
+            fi
+        done
+
+        OUTPUT_PATH=$(params.subdirectory)/src_dest_combination.json
+        echo -n "$OUTPUT_PATH" > "$(results.destination_path.path)"
+        echo "$json_output" | tee "$(workspaces.data.path)/$OUTPUT_PATH"
+    - name: run-exodus-command
+    # TODO: Remove after testing
+      image: quay.io/distribution_relengs/publish-to-dev-portal-test-image
+      # Under construction
+      # TODO: Update this to do joined publish
+      # https://github.com/release-engineering/exodus-rsync?tab=readme-ov-file#joined-publish
+      script: |
+        #!/bin/bash
+        set -e       
+        conf_dir="$(workspaces.data.path)/$(params.subdirectory)/exodus-rsync.conf"
+        src_dest_json="$(workspaces.data.path)/$(params.subdirectory)/src_dest_combination.json"
+        cat "$src_dest_json" | jq -r 'to_entries[] | "\(.key) \(.value)"' | while read -r src dest; do
```

Comment:
I'm not sure how this is supposed to work. Currently this will not not exist. The previous step does not write into this file. Instead, it only writes this file path to `$(results.destination_path.path)` - which seems confusing, because I would expect `$destination_path` to be written there instead.

Reply:
But also, the only result this task declares is `srcDestPath`.

Reply:
I thought I was writing content to that location on lines 115-117 and accessing it here similar to this (https://github.com/konflux-ci/release-service-catalog/blob/development/tasks/collect-data/collect-data.yaml#L173)

Regarding `srcDestPath`, I updated that on line 116.

Trying to store key:value for src:dest which can be accessed here 
and another task (publish-to-cgw) later.

Reply:
OK, my bad, you're right you're writing to that file on line 117. Could both steps maybe use the same variable for that file? It would increase readability. One uses `OUTPUT_PATH`, but it's only the relative path inside the workspace, and the other uses `src_dest_json`, but it's the absolute path.

I thought I saw an instance of saving stuff to a result that doesn't exist, but maybe I was seeing things that didn't exist...

Reply:
Updated as such, ty!

I did change that on you in my last commit :)
https://github.com/konflux-ci/release-service-catalog/compare/0b90fc923c6754cf3456e75367fec25f04bcdf3b..294ae36286ad403cd874ad1fc8016b19fb1ccbad

Reply:
ok, cool!